### PR TITLE
fetchMavenArtifact: deprecate phases & use pname+version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes

#28910 & #103997

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
I'm not actually familiar with `fetchMavenArtifact`, only modified it once to unbreak a package.

File: pkgs/build-support/fetchmavenartifact/default.nix [hunk 1]
```diff
@@ -37,13 +37,13 @@ assert (url == "") || (urls == []);
...
-  name_ =
+  pname =
     lib.concatStrings [
-      (lib.replaceStrings ["."] ["_"] groupId) "_"
-      (lib.replaceStrings ["."] ["_"] artifactId) "-"
-      version
+      (lib.replaceStrings [ "." ] [ "_" ] groupId)
+      "_"
+      (lib.replaceStrings [ "." ] [ "_" ] artifactId)
     ];
```
Comment (SuperSandro2000) [inline pkgs/build-support/fetchmavenartifact/default.nix:45]:
```suggestion
  pname = (lib.replaceStrings [ "." ] [ "_" ] groupId) + "_" + (lib.replaceStrings [ "." ] [ "_" ] artifactId);
```
Is this equal to this?

Comment (Stunkymonkey) [reply pkgs/build-support/fetchmavenartifact/default.nix:45]: looks to me equal yes. i will apply it.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 219707` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>liquibase</li>
    <li>liquibase_redshift_extension</li>
    <li>opentsdb</li>
    <li>postgresql_jdbc</li>
    <li>redshift_jdbc</li>
    <li>sqlite-jdbc</li>
    <li>squirrel-sql</li>
  </ul>
</details>

**Pull request merged**